packages/eas-cli/src/vcs/clients/git.ts

```diff
@@ -386,3 +433,51 @@ async function setGitCaseSensitivityAsync(
     });
   }
 }
+
+async function assertEnablingGitCaseSensitivityDoesNotCauseNewUncommittedChangesAsync(
+  cwd: string
+): Promise<void> {
+  // Remember uncommited changes before case sensitivity change
+  // for later comparison so we log to the user only the files
+  // that were marked as changed after the case sensitivity change.
+  const uncommittedChangesBeforeCaseSensitivityChange = await gitStatusAsync({
+    showUntracked: true,
+    cwd,
+  });
+
+  const isCaseSensitive = await isGitCaseSensitiveAsync(cwd);
+  await setGitCaseSensitivityAsync(true, cwd);
+  try {
+    const uncommitedChangesAfterCaseSensitivityChange = await gitStatusAsync({
+      showUntracked: true,
+      cwd,
+    });
+
+    if (
+      uncommitedChangesAfterCaseSensitivityChange !== uncommittedChangesBeforeCaseSensitivityChange
+    ) {
```

Comment:
what is the format of these 2? is there a chance that git will return files in different order causing us to enter this if statement even if nothing changed? maybe we should operate on sets here?